ebos/eclwriter.hh

```diff
@@ -328,8 +328,15 @@ public:
                          blockData,
                          aquiferData);
 
-            const auto& udq_config = schedule().getUDQConfig(reportStepNum);
-            udq_config.eval( reportStepNum, schedule().wellMatcher(reportStepNum), summaryState(), udqState() );
+            /*
+              Off-by-one-fun: The reportStepNum argument corresponds to the
+              report step these results will be written to, whereas the argument
+              to UDQ function evaluation corresponds to the report step we are
+              currently on.
+            */
+            auto udq_step = reportStepNum - 1;
```

Comment:
I guess there's a guarantee from the broader context that `reportStepNum > 0` here, but I think we should add a comment to that effect to assist future readers.